src/ashley/models.py

```diff
@@ -139,6 +140,9 @@ def sync_user_groups(self, user, roles: List[str]) -> None:
         """
         Synchronize the group membership of a user for this LTI Context
         """
+        # Control user is part of this lti_context
+        if user.lti_consumer_id != self.lti_consumer_id:
+            raise PermissionDenied(_("user must be part of this lti context"))
```

Comment:
In fact, we call this function to link a user to a specific LTI context.
I suggest the following error instead : 

```suggestion
            raise PermissionDenied(_("The User and LTIContext must be part of the same LTI Consumer"))
```